tests/phpunit/tests/fonts/wpDefaultFontDir.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Test wp_default_font_dir().
+ *
+ * @package WordPress
+ * @subpackage Font Library
+ *
+ * @group fonts
+ * @group font-library
+ *
+ * @covers ::wp_default_font_dir
+ */
+class Tests_Fonts_WpDefaultFontDir extends WP_UnitTestCase {
+
+	public function test_default_font_dir() {
+		$font_dir = wp_default_font_dir();
+
+		$this->assertSame(
+			$font_dir,
+			array(
+				'path'    => WP_CONTENT_DIR . '/fonts',
+				'url'     => content_url( 'fonts' ),
+				'subdir'  => '',
+				'basedir' => WP_CONTENT_DIR . '/fonts',
+				'baseurl' => content_url( 'fonts' ),
+				'error'   => false,
+			),
+			'The font directory should be a dir inside wp-content'
+		);
+	}
+
+	public function test_uploads_font_dir() {
+		$font_dir = wp_default_font_dir( true );
+		chmod( $font_dir['path'], 0000 );
```

Comment:
I haven't found a way to restrict the writing permissions in the context of unit tests. This `chmod` line isn't working. It is there just to show how the functionality could be tested.

Reply:
No idea how to test this either. If the option is changed to record the fallback rather than the directory structure then you migh be able to filter it using the `pre_option_{$option}` hook.